app/controllers/home_controller.rb

```diff
@@ -1,6 +1,28 @@
+require 'rss'
+require 'open-uri'
+
 class HomeController < ActionController::Base
   layout 'application'
 
   def index
+    @status = Rails.cache.read("rave_status")
   end
+
+  def send_alert
+    if request[:status] == "on"
+      status = "off"
+    else
+      status = "on"
+    end
+
+    Rails.cache.write("rave_status", status)
```

Comment:
It might be clearer if cache duration was explicitly set here with explanation that alert will always default no-emergency state after it expired.